modules/aui2s/aui2s.c

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file aui2s freeRTOS I2S audio driver module
+ *
+ * Copyright (C) 2019 cspiel.at
+ */
+#include <sys/types.h>
+#include <sys/time.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "freertos/FreeRTOS.h"
+#include <re.h>
+#include <rem.h>
+#include <baresip.h>
+
+#include "driver/i2s.h"
+#include "aui2s.h"
+
+
+static struct ausrc *ausrc = NULL;
+static struct auplay *auplay = NULL;
+
+/*---------------------------------------------------------------
+                            CONFIG
+---------------------------------------------------------------*/
+
+static int aui2s_init(void)
+{
+	int err;
+
+	err  = ausrc_register(&ausrc, baresip_ausrcl(),
+			      "aui2s", aui2s_src_alloc);
+	err |= auplay_register(&auplay, baresip_auplayl(),
+			       "aui2s", aui2s_play_alloc);
+
+
+	return err;
+}
+
+
+static enum I2SOnMask _i2s_on = I2O_NONE;
+
+
+int aui2s_start(uint32_t srate, enum I2SOnMask playrec)
+{
+	esp_err_t err;
+	bool start = _i2s_on == I2O_NONE;
+	_i2s_on = _i2s_on | playrec;
+
+	if (srate * 4 % DMA_SIZE) {
+		warning("aui2s: srate*4 % DMA_SIZE != 0\n");
+		return EINVAL;
+	}
+
+	info("%s start with _i2s_on=%d", __func__, _i2s_on);
+	if (start) {
+		i2s_config_t i2s_config = {
+			.mode = I2S_MODE_MASTER | I2S_MODE_RX | I2S_MODE_TX,
+			.sample_rate =  srate,
+			.bits_per_sample = 32,
+			.communication_format = I2S_COMM_FORMAT_I2S |
+				I2S_COMM_FORMAT_I2S_MSB,
```

Comment:
is it possible to use a communication format that is compatible
with the input from baresip core, i.e. signed 16-bit PCM ?

in this case, we can get rid of the sample format conversion...